pkg/controller/operators/catalog/not_found.go

```diff
@@ -0,0 +1,46 @@
+package catalog
+
+import (
+	"fmt"
+
+	operatorsv1alpha1 "github.com/operator-framework/api/pkg/operators/v1alpha1"
+
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/discovery"
+)
+
+// gvkNotFoundError is returned from installplan execution when a step contains a GVK that is not found on cluster.
+type gvkNotFoundError struct {
+	gvk  schema.GroupVersionKind
+	name string
+}
+
+func (g gvkNotFoundError) Error() string {
+	return fmt.Sprintf("api-server resource not found installing %s %s: GroupVersionKind %s not found on the cluster. %s", g.gvk.Kind, g.name, g.gvk.String(),
+		"This API may have been deprecated and removed, see https://kubernetes.io/docs/reference/using-api/deprecation-guide/ for more information.")
+}
+
+// notFound queries discovery to check supported APIs on the server.
+func notFound(client discovery.DiscoveryInterface, stepResource operatorsv1alpha1.StepResource) error {
+	resourceList, err := client.ServerResourcesForGroupVersion(schema.GroupVersion{Group: stepResource.Group, Version: stepResource.Version}.String())
```

Comment:
nit: I think it would be cleaner if this function either has a `schema.GroupVersionKind` parameter or declares only one up front. The type has some methods that make extracting the `GroupVersion` straight forward too: https://github.com/kubernetes/apimachinery/blob/a644435e2c133d990b624858d9c01985d7f59adf/pkg/runtime/schema/group_version.go#L157